### PR TITLE
EN-63554: slack to teams

### PR DESCRIPTION
**Pull request by sarah-salinas**
We recently migrated from slack to teams for our messaging service. This commit updates the build failure messages from slack to teams.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,97 +1,67 @@
...
     label params.AGENT
   }
-
   triggers {
     issueCommentTrigger('^retest$')
   }
-
   environment {
     SERVICE = 'image-resize-service'
-    RUNNING_IN_CI = 'true'
```
Comment (sarah-salinas) [inline Jenkinsfile:26]: None of these environment variables I deleted are needed

File: Jenkinsfile [hunk 2]
```diff
@@ -1,97 +1,67 @@
...
-    REPOSITORY_OWNER = "${env.GIT_URL.tokenize('/')[2]}"
     SERVICE_VERSION = '0.0.1'
-    ARTIFACTORY_CREDS = credentials('shared-eng-artifactory-creds')
   }
-
   stages {
     stage('Pull Request') {
       when { changeRequest() }
-      stages {
-        stage('Build') {
```
Comment (sarah-salinas) [inline Jenkinsfile:38]: I removed the nested stage and the script tag

File: Jenkinsfile [hunk 3]
```diff
@@ -1,97 +1,67 @@
...
-
-    stage('Master') {
+    stage('Build') {
       when { branch 'main' }
       stages {
         stage('Build and push image') {
           steps {
-            sh 'sbt assembly'
+            script {
+              lastStage = env.STAGE_NAME
```
Comment (sarah-salinas) [inline Jenkinsfile:38]: I am inserting this lastStage variable to consolidate messages but still report the stage.  In order to do so, it needs to go in a script tag, so I've just put everything in one script tag for this stage.

File: Jenkinsfile [hunk 4]
```diff
@@ -1,97 +1,67 @@
...
-            // this must match the "RUBY VERSION" in https://github.com/socrata/apps-marathon/blob/main/Gemfile.lock
-            withRbenv("2.5.3") {
-              script {
-                marathon.checkoutAndInstall()
-                marathon.deploy(env.SERVICE, 'staging', env.DOCKER_TAG)
-              }
+            script {
+              lastStage = env.STAGE_NAME
+              // Deploys env.SERVICE with tag env.DOCKER_TAG to staging by default
+              marathonDeploy()
```
Comment (sarah-salinas) [inline Jenkinsfile:56]: Simpler syntax

File: Jenkinsfile [hunk 5]
```diff
@@ -1,97 +1,67 @@
...
-            }
-            failure {
-              slackSend(channel: "${env.SLACK_CHANNEL}", color: 'RED', message: "${env.SERVICE} (${env.DOCKER_TAG}): Failed deploying to staging marathon - ${env.BUILD_URL}")
-            }
+        }
+      }
+      post {
+        failure {
+          script {
+            teamsMessage(message: "Build [${currentBuild.fullDisplayName}](${env.BUILD_URL}) has failed in stage ${lastStage}", webhookCredentialID: WEBHOOK_ID)
```
Comment (sarah-salinas) [inline Jenkinsfile:64]: Note that this post is on the 'Build' stage not the 'PR' stage so PR failures will not be reported.

**Comment by sarah-salinas** [comment]
## Testing

I made the PR build the main build stage and caused it to fail.  The message posted as expected ✅

**Review by aynleslie — APPROVED** [review]
thank you.

**Pull request merged**